### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by rishbitsnbytes**
Added: Authentication!
Kindly review!

### CODE REVIEW

File: src/pages/AuthPages/LoginPage.jsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    try {
+      const { data } = await loginService(formData);
+
+      const { encodedToken, foundUser } = data;
+      authDispatch({
+        action: {
+          type: "SET_AUTH_LOADING",
+          payload: { authLoading: true },
+        },
+      });
```
Comment (TejasShekar) [inline src/pages/AuthPages/LoginPage.jsx:63]: If this is the dispatch that handles auth loading, push it outside, right before the try block. This way, the loading happens until you get data from backend and then you can make the next dispatch to initiate the auth.

**Review by ashishkatiyar2404 — APPROVED** [review]
Well done rishab nice code .. functionality looks great
Looks good to merge

**Review by TejasShekar — CHANGES_REQUESTED** [review]
Authentication does not seem to be working at the moment.
Code looks good.

File: src/pages/AuthPages/LoginPage.jsx (hunk 1 above)
Comment (rishbitsnbytes) [reply src/pages/AuthPages/LoginPage.jsx:63]: Noted! Thanks!

**Pull request merged**